README.md

```diff
@@ -8,8 +8,58 @@ The framework itself comprises two components:
 - driver - responsible for controlling the workspace and IDE startup
 - probe - used to interact with the IDE  
 
+#### Overview
 
-#### Configuration
+A single test case consists of a configuration and workflow specification. 
+Configuration can either be:
+
+A) loaded from file,
+ ```scala
+private val config = Config.fromClasspath("path/to/file")
+
+@Test def test = IntelliJFixture.fromConfig(config).run {intelliJ => ???}
+```
+B) provided as a string, or
+```scala
+private val config = Config.fromString("""probe { workspace.path = /foo/bar } """)
+@Test def test = IntelliJFixture.fromConfig(config).run {intelliJ => ???}
+
+```
+C) specified programmatically.
+```scala
+private val fixture = IntelliJFixture(
+  workspaceTemplate = WorkspaceTemplate.fromFile(path),
+  version = IntelliJVersion("202.5792.28-EAP-SNAPSHOT"),
+  plugins = List(Plugin("org.intellij.scala", "2020.2.7"))
+)
+
+@Test def test = fixture.run {intelliJ => ???} 
+```
+
+Workflow can only be defined programmatically, since it comprises sequence of intertwined:
+1. probe commands,
+2. IDE state queries,
+3. workspace manipulation,
+4. custom verification logic.
+
+```scala
+@Test def test = fixture.run { intelliJ =>
+  val buildSbt = intelliJ.workspace.resolve("build.sbt")
+  Files.write(buildSbt, ???)  
```

Comment:
I think it would be nicer to fill in `???` everywhere to make examples complete. Here, probably simply `name := "foo"` would do (?)